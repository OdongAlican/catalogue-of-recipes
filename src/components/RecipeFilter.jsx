import React from 'react'

function RecipeFilter ({ recipiesData, filterRecipe }){
    const handleFilter = ({ target }) => {
        filterRecipe(target.value)
    }

    return (
        <div>
            <select onChange={handleFilter}>
                <option value="All" key="All">
                    All
                </option>
                {
                    recipiesData.recipies.map( recipe => (
                        <option key={ recipe.id } value = { recipe.userId }> { recipe.userId } </option>
                    ))
                }
            </select>
        </div>
    )
}

  export default RecipeFilter