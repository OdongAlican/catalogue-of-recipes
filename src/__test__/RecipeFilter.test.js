import React from 'react'
import RecipeFilter from '../components/RecipeFilter'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

afterEach(cleanup)
const initialState = {output:10}
const mockStore = configureStore()
let store

const recipiesData = {
    recipies: [{
    image:"sample", 
    title:"title",
    site: "site",
    publishedDate: "2020-09-23 15:00:00"
}]}

it("renders RecipeFilter correctly", ()=>{
    store = mockStore(initialState)
    const { getAllByTestId } = render(<Provider store={store}><RecipeFilter recipiesData = { recipiesData }/></Provider>)
    const result = getAllByTestId('recipeFilter')
    expect(result[0]).toHaveTextContent("SELECT COMPANY:All")
})

it("another rendering of nav-bar correctly", ()=>{
    store = mockStore(initialState)
    const { getAllByTestId } = render(<Provider store={store}><RecipeFilter recipiesData = { recipiesData }/></Provider>)
    const result = getAllByTestId('recipeFilter')
    expect(result[0]).not.toHaveTextContent("some random string")
})

it('matches RecipeFilter snapshot', ()=> {
    const tree = renderer.create(<Provider store={store}><RecipeFilter recipiesData = { recipiesData }/></Provider>).toJSON()
    expect(tree).toMatchSnapshot();
})