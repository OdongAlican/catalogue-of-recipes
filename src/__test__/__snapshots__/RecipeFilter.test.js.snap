// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches RecipeFilter snapshot 1`] = `
<div
  className="bg-secondary my-2 p-2 d-flex justify-content-center align-items-center"
  data-testid="recipeFilter"
>
  <label
    className=" text-white mr-1 mt-1 font-weight-bold"
    htmlFor="filter"
  >
    SELECT COMPANY:
  </label>
  <select
    className="form-control col-md-3 p-1"
    onChange={[Function]}
  >
    <option
      value="All"
    >
      All
    </option>
    <option>
       
       
    </option>
  </select>
</div>
`;
